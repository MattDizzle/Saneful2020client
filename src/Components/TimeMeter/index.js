import React from 'react';

import './TimeMeter.scss';

const TimeMeter = (props) => {

  const renderDayAndTime = () => {

    let timeLeft = props.currentTime;

    const numDays = Math.floor(timeLeft / 1440);
    timeLeft -= (1440 * numDays);
    const numHours = Math.floor(timeLeft / 60);
    timeLeft -= (60 * numHours);
    const numMins = timeLeft;

    const hourString = numHours.toString().padStart(2, '0');
    const minString = numMins.toString().padStart(2, '0');

    return (
      <div>
        <p>{`Day ${numDays}`}</p>
        <p>{`${hourString}:${minString}`}</p>
      </div>
    );

  };

  return (
    <>
      {renderDayAndTime()}
    </>
  );
};

export default TimeMeter;