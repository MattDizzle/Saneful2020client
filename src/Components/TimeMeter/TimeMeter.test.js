import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import TimeMeter from './index';


describe(`TimeMeter Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <TimeMeter />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <TimeMeter />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})