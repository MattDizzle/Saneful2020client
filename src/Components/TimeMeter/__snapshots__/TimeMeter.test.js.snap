// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeMeter Component Snapshot test Renders the UI as expected 1`] = `
<div>
  <p>
    Day NaN
  </p>
  <p>
    NaN:NaN
  </p>
</div>
`;
